### ISSUE TITLE
ListView with RecyclerViewBackedScrollView crash on Android

### ISSUE THREAD

**Issue by joles86** [facebook/react-native#10560]
### Description

When using `renderScrollComponent={props => <RecyclerViewBackedScrollView {...props} />` on a ListView, the app crash on android giving this error :

```
java.lang.AbstractMethodError: abstract method "boolean android.support.v7.widget.RecyclerView$ItemAnimator.animateAppearance(android.support.v7.widget.RecyclerView$ViewHolder, android.support.v7.widget.RecyclerView$ItemAnimator$ItemHolderInfo, android.support.v7.widget.RecyclerView$ItemAnimator$ItemHolderInfo)"
at android.support.v7.widget.RecyclerView.animateAppearance(RecyclerView.java:3528)
at android.support.v7.widget.RecyclerView$4.processAppeared(RecyclerView.java:461)
at android.support.v7.widget.ViewInfoStore.process(ViewInfoStore.java:249)
at android.support.v7.widget.RecyclerView.dispatchLayoutStep3(RecyclerView.java:3385)
at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3135)
at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3568)
at android.view.View.layout(View.java:16630)
at android.view.ViewGroup.layout(ViewGroup.java:5437)
at com.facebook.react.uimanager.NativeViewHierarchyManager.updateLayout(NativeViewHierarchyManager.java:192)
at com.facebook.react.uimanager.NativeViewHierarchyManager.updateLayout(NativeViewHierarchyManager.java:177)
at com.facebook.react.uimanager.UIViewOperationQueue$UpdateLayoutOperation.execute(UIViewOperationQueue.java:123)
at com.facebook.react.uimanager.UIViewOperationQueue$2.run(UIViewOperationQueue.java:782)
at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:829)
at com.facebook.react.uimanager.UIViewOperationQueue.access$1500(UIViewOperationQueue.java:44)
at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:868)
at com.facebook.react.uimanager.GuardedChoreographerFrameCallback.doFrame(GuardedChoreographerFrameCallback.java:32)
at com.facebook.react.uimanager.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:131)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:856)
at android.view.Choreographer.doCallbacks(Choreographer.java:670)
at android.view.Choreographer.doFrame(Choreographer.java:603)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```
### Reproduction

```
      <ListView
          dataSource={this.state.eventsDataSource}
          pageSize={this.state.eventsCount}
          ref="eventsListView"
          refreshControl={
            <RefreshControl
              enabled={false}
              refreshing={this.props.fetchingEvents}
            />
          }
          renderRow={(event) =>
            <EventListItem
              color={this.getEventColor(event)}
              event={event}
              height={EVENT_LIST_ITEM_HEIGHT}
              onPress={() => {}}
            />
          }
          renderScrollComponent={props => <RecyclerViewBackedScrollView {...props} />}
          renderSectionHeader={(sectionData, date) =>
            <View style={styles.sectionHeader}>
              <Text style={styles.sectionHeaderText}>{I18n.l('date.formats.long_day_month_date_year', date)}</Text>
            </View>
          }
        />
```
### Additional Information
- React Native version: 0.35
- Platform: Android
- Operating System: MacOS

**Comment by IbrahimAmin** [comment]
I confirm it also happened with me

**Comment by rborn** [comment]
same here for React 0.37

**Comment by hramos** [comment]
Can you reproduce this on [rnplay.org](https://rnplay.org/)?

**Comment by rborn** [comment]
Mmm, was using Nuclide,and somehow I think the packager went bonkers.
I will come back to you if I manage to crash it with normal start from cli.

**Comment by charlesluo2014** [comment]
same for RN 0.37

**Comment by chang-chaunce** [comment]
I have the same problem.

com.android.support:appcompat-v7:23.0.1
com.android.support:design:23.0.1
compile 'com.android.support:recyclerview-v7:23.0.1
can work well.

if recyclerview or appcompat version > 23.0.1 ,will crash.

same for RN 0.37-0.39

Additional Information:
    React Native version: 0.39
    Platform: Android
    Operating System: MacOS

**Comment by rborn** [comment]
Not sure the recycler is used (and therefore maintained anymore 😢 ) ->  https://github.com/facebook/react-native/pull/11095

**Comment by tychota** [comment]
I've reopened a PR to depreciate this.

**Comment by teodors** [comment]
@tychota Why do you want to deprecate it? Many people use RecyclerViewBackedScrollView and there is no alternative. Just search issue section for "RecyclerViewBackedScrollView" to see just how many.

**Comment by tychota** [comment]
@teodors : https://github.com/facebook/react-native/pull/11445

tl;dr

RecyclerViewBackedScrollView seems to be an experiment, not maintained by Facebook anymore (thus breaking in last RN version).

It was useful in the Past cause it was a great workaround around ListView bugs and perf issues but now:
- bug where insertion of items make the list bounce back is fixed in list view
- the performance benefit is really minimal not that `removeClippedSubviews` exist in ListView (according to core contributor discussion, I have no benchmark)
- no one is left to maintain it and push it so it replace list view

**Comment by tychota** [comment]
@teodors: Well I may be wrong, so please comment on the pull request that you don't want it to be merged.

The initial PR was about to get merged to depreciate this, but a member of RN core team decided to extract it to a repo before. That as never the case, as far as I know and now the RecyclerViewBackedScrollView make android crash.

**Comment by sibelius** [comment]
@teodors u can use this https://github.com/facebook/react-native/blob/master/Libraries/Experimental/WindowedListView.js

**Comment by hramos** [comment]
Closing as ListView is deprecated in 0.43. Use FlatList.
